nexus/src/app/instance.rs

```diff
@@ -898,6 +925,24 @@ impl super::Nexus {
             &requested,
         )? {
             InstanceStateChangeRequestAction::AlreadyDone => Ok(()),
+            InstanceStateChangeRequestAction::UpdateRuntime(new_runtime) => {
+                let instance_id =
+                    InstanceUuid::from_untyped_uuid(prev_instance_state.id());
+                let changed = self
+                    .datastore()
+                    .instance_update_runtime(&instance_id, &new_runtime)
+                    .await
+                    .map_err(InstanceStateChangeError::Other)?;
+                if !changed {
+                    // TODO(eliza): perhaps we should refetch the instance here
+                    // and return Ok if it was in the desired state...
+                    Err(InstanceStateChangeError::Other(Error::conflict(
+                        "instance changed state before it could be updated",
+                    )))
```

Comment:
i don't love this error message but couldn't come up with something better --- any suggestions?

Reply:
This one's tricky because we don't really want the error to start talking about details like "generation numbers" or "optimistic concurrency control," but it's hard to explain what's really happening without bringing those concepts in. Would being more verbose help? For example:

> The instance was previously in a state that allowed the requested state change, but the instance's state changed before the request could be completed.

Reply:
I changed it to use that, thanks for the suggestion!